pkg/plugins/pluginrpc-gen/README.md

```diff
@@ -0,0 +1,68 @@
+Plugin RPC Generator
+====================
+
+Generates go code from a go interface definition for proxying between the plugin
+API and the subsystem being extended.
+
+## Usage
+
+Given an interface definition:
+
+```go
+type volumeDriver interface {
+	Create(name string, opts opts) (err error)
+	Remove(name string) (err error)
+	Path(name string) (mountpoint string, err error)
+	Mount(name string) (mountpoint string, err error)
+	Unmount(name string) (err error)
+}
+```
+
+**Note** All function options and return values must be named in the definition.
+
+Run the generator:
+
+```bash
+$ pluginrpc-gen -type volumeDriver -name VolumeDriver -i volumes/drivers/extpoint.go -o volumes/drivers/proxy.go
+```
+
+Where:
+- `-type` is the name of the interface to use
+- `-name` is the subsystem that the plugin "Implements"
+- `-i` is the input file containing the interface defintiion
+- `-o` is the output file where the the generated code should go
+
+**Note** The generated code will use the same package name as the one defined in the input file
+
+Optionally, you can skip functions on the interface should should not be
+implemented in the generated proxy code by passing in the function name to `-skip`.
```

Comment:
Just something I noticed; Does this use single dashes for long options? (`-skip`, not `--skip`); would be more consistent with "Docker" to use double dashes.